### ISSUE TITLE
mypy-nonfloat-decimal broken on `release-0.960`

### ISSUE THREAD

**Issue by rouge8** [python/mypy#12861]
**Bug Report**

[mypy-nonfloat-decimal](https://github.com/yedpodtrzitko/mypy-nonfloat-decimal) no longer works on https://github.com/python/mypy/commit/128661ccbf4396ed1538421c481f1773c7490169. Mentioned on https://github.com/python/mypy/issues/12807#issuecomment-1135278105, but was told to open an issue. :) I also have an internal plugin using `get_dynamic_class_hook()` that is no longer working, but that's probably a separate issue and harder to breakdown.

**To Reproduce**

(Write your steps here:)

1. `pip install mypy-nonfloat-decimal https://github.com/python/mypy/archive/128661ccbf4396ed1538421c481f1773c7490169.zip#egg=mypy`
2. `t.py`:
  ```python
  from decimal import Decimal

  Decimal(1.2)
  ```
3. `mypy.ini`:
  ```ini
  [mypy]
  plugins = mypy_nonfloat_decimal
 ```
4. `mypy t.py`

**Expected Behavior**

```
t.py:3: error: Invalid type passed to Decimal (expected "Union[int, str, Decimal]"), got builtins.float instead (offender: float)
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**
```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 128661ccbf4396ed1538421c481f1773c7490169
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.10.4
- Operating system and version: macOS 12.3.1

**Comment by JelleZijlstra** [comment]
This is most likely because of https://github.com/python/typeshed/pull/7640. The plugin needs to look for `_decimal.Decimal` instead.

**Comment by rouge8** [comment]
Ah you're right, that one's easy. I'll try and reproduce the internal plugin that no longer seems to work tonight or tomorrow.
